File: benches/listrev/listrev.ml
let rec range_tailrec (n : int) (acc : int list) =
    if n == 0 then acc
              else range_tailrec (n - 1) (n :: acc)

let rec rev (l : int list) (acc : int list) =
    match l with
    | [] -> acc
    | hd :: tl -> rev tl (hd :: acc)

let main =
    for _ = 1 to 500 do
        let ls = range_tailrec 10_000 [] in
        let r = rev ls [] in
        Stdlib.print_int (List.length r)
    done

File: benches/listrev/listrev.py
import sys
sys.setrecursionlimit(50_000)

def ll_range(n, acc):
    if n == 0:
        return acc
    else:
        return ll_range(n - 1, (n, acc))

def ll_reverse(l, acc):
    match l:
        case (): 
            return acc
        case (hd, tl):
            return ll_reverse(tl, (hd, acc))

for _ in range(500):
    ls = ll_range(10_000, ())
    rev = ll_reverse(ls, ())
